src/main/java/org/opentripplanner/routing/impl/OrcaFareServiceImpl.java

```diff
@@ -193,22 +247,22 @@ private static RideType classify(Route routeData, String tripId) {
      * Define which discount fare should be applied based on the fare type. If the ride type is unknown the discount
      * fare can not be applied, use the default fare.
      */
-    private float getLegFare(Fare.FareType fareType, RideType rideType, float defaultFare, Route route) {
+    private float getLegFare(Fare.FareType fareType, RideType rideType, float defaultFare, Ride ride) {
         if (rideType == null) {
             return defaultFare;
         }
         switch (fareType) {
             case youth:
             case electronicYouth:
-                return getYouthFare(fareType, rideType, defaultFare, route);
+                return getYouthFare(fareType, rideType, defaultFare, ride.routeData);
             case electronicSpecial:
                 return getLiftFare(rideType, defaultFare);
             case electronicSenior:
             case senior:
-                return getSeniorFare(fareType, rideType, defaultFare, route);
+                return getSeniorFare(fareType, rideType, defaultFare, ride.routeData);
             case regular:
             case electronicRegular:
-                return getRegularFare(fareType, rideType, defaultFare, route);
+                return getRegularFare(fareType, rideType, defaultFare, ride.routeData, ride);
```

Comment:
Passing `ride.routeData` seems redundant if you pass `ride ` also, so remove the `Route` argument.